Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1402,28 +1418,41 @@ WebIDL::ExceptionOr<void> Navigable::navigate(NavigateParams params)
 
     auto source_document = params.source_document;
     auto exceptions_enabled = params.exceptions_enabled;
-
+    auto& url = params.url;
+    auto user_involvement = params.user_involvement;
     auto& active_document = *this->active_document();
     auto& realm = active_document.realm();
     auto& page_client = active_document.page().client();
 
     // AD-HOC: If we are not able to continue in this process, request a new process from the UI.
-    if (is_top_level_traversable() && !page_client.is_url_suitable_for_same_process_navigation(active_document.url(), params.url)) {
-        page_client.request_new_process_for_navigation(params.url);
+    if (!page_client.is_url_suitable_for_same_process_navigation(active_document.url(), url)) {
```

Comment:
Merge conflict? We don't want to lose the `is_top_level_traversable` check. See 00aa6203ff04325a229f6199c49a5ee84ab63c75.

Reply:
oops yep, I had to do some gymnastics to rebase this, given your change and Shannon's change to make two of the types affected no longer take a realm when allocating

Reply:
Sorry, that came up when I was looking at that static function on Document function as well 😅  I also have this commit sitting around if helpful by the way: https://github.com/shannonbooth/ladybird/commit/c09aa7a

Reply:
At yeah that's the fetch spec algo that Domenic was updating due to my finding about the "request's window" needing updated for this.

Reply:
Right just a catchup commit to make the actual spec changes easier to see/apply